File: Python/programmers/etc/level1/바탕화면_정리.py
def solution(wallpaper):
    lux, rdx, luy, rdy = 50, 0, 50, 0
    r, c = len(wallpaper), len(wallpaper[0])
    
    for i in range(r):
        for j in range(c):
            if wallpaper[i][j] == "#":
                lux = min(lux, i)
                rdx = max(rdx, i+1)
                luy = min(luy, j)
                rdy = max(rdy, j+1)
                
    return [lux, luy, rdx, rdy]


wallpaper = [".#...", "..#..", "...#."]
print(solution(wallpaper)) # [0, 1, 3, 4]

wallpaper = ["..........", ".....#....", "......##..", "...##.....", "....#....."]
print(solution(wallpaper)) # [1, 3, 5, 8]

wallpaper = [".##...##.", "#..#.#..#", "#...#...#", ".#.....#.", "..#...#..", "...#.#...", "....#...."]
print(solution(wallpaper)) # [0, 0, 7, 9]

wallpaper = ["..", "#."]
print(solution(wallpaper)) # [1, 0, 2, 1]

File: Java/programmers/etc/level1/바탕화면정리.java
package CodeTest.Java.programmers.etc.level1;

public class 바탕화면정리 {
    public int[] solution(String[] wallpaper) {
        int lux = 50, rdx = 0, luy = 50, rdy = 0;
        int r = wallpaper.length, c = wallpaper[0].length();

        for (int i=0; i<r; i++) {
            for (int j=0; j<c; j++) {
                if (wallpaper[i].substring(j, j+1).equals("#")) {
                    lux = Math.min(lux, i);
                    rdx = Math.max(rdx, i+1);
                    luy = Math.min(luy, j);
                    rdy = Math.max(rdy, j+1);
                }
            }
        }

        return new int[]{lux, luy, rdx, rdy};
    }
}
